Port various changes from the Arc Theme

Kogiku opened the pull request:
Arc changes for reference and credit: https://github.com/arc-design/arc-theme/compare/20171105...c434a56d2cb5ee49969a62a462db380dee8be62a
  * [Changes ported from https://github.com/{arc-design,jnsh}/arc-theme]
    - build
      + Compile with sassc and removed autogenrated css files
      + Added plank theme to makefile (option --disable-plank)
      + Separate gtk3 and gnome-shell versioning (options --with-gnome-shell= --with-gtk3=)
    - Cinnamon
      + Remove margin:auto from cinnamon menu applet
    - GNOME Shell
      + Remove no longer supported versions
      + Add support for versions 3.24 to 3.34
      + [3.24+] Shell overview caption fix
      + [3.24+] Remove message-list-section clear button
      + [3.24+] Remove message-list-section-title definitions
      + [3.24+] Add Pad OSD theming
      + [3.26+] Fix nesting of elements
      + [3.26+] Use font-relative width for prompts
      + [3.26+] loginDialog: Only highlight a single item at any time
      + [3.26+] More compact search results
      + [3.26+] Update mount dialog classes
      + [3.26+] Add message dialog style classes
      + [3.26+] Use .message-dialog-* classes
      + [3.26+] Add extension-dialog style classes
      + [3.26+] Add inhibit-shortcuts-dialog class
      + [3.26+] Reduce window preview spacing
      + [3.26+] Remove .legacy-tray class
      + [3.26+] Add system action icon theming
      + [3.28+] Align system-sub-menus with parent
      + [3.28+] Update OSK theme
      + [3.30+] Fix gnome sliders and add overdrive capability
      + [3.30+] Fix overview page indicators
      + [3.30+] Round user icons in GNOME dialogs
      + [3.30+] Update login user-list 
      + [3.32+] fix search results not displayed when exiting lock screen
      + [3.32+] Replace page indicator & corner ripple SVGs with CSS
      + [3.32+] Add select-area-rubberband theming 
      + [3.34+] Add new prompt-dialog classes
      + [3.34+] Add folder rename popup styling 
      + [3.34+] Add drag-over style for app folders
    - GTK3
      + Remove no longer supported versions
      + [3.18+] Resolve MATE panel icon padding
      + [3.20+] Tweak nautilus 3.30 pathbar
      + [3.20+] Fix horizontal stepper icons
      + [3.20+] Fix vertical and horizontal levelbar values
      + [3.20+] Infobar should respect message type closes
    - Metacity
      + add buttons to modal geometry
  * [Changes/fixes by Kogiku]
    - GNOME Shell
      + [3.24+] theme weather button
      + [3.32+] fix activities overview close button
      + [3.32+] update switch buttons

Kogiku committed:
Ported various changes from https://github.com/arc-design/arc-theme

Kogiku committed:
AUTHORS: tab

Kogiku committed:
README.md: undo removal of 'Gnome-Shell-3.16'

oberon-manjaro commented:
Thank you. This is quite an impressive PR! :wink: 
I will test the changes very soon.

Kogiku committed:
Re-add deleted file & bump max gnome requirement

Re-add deleted extra/meld/* (oops) README.md: bump max gnome requirement to 3.34

Kogiku committed:
README.md: undo removal of 'Vertex-'

Kogiku commented:
Sorry about all the commit pollution. I probably should have force merged the first commit (still learning).

I think the only other nitpicky thing that might need changing is the version dates in `configure.ac`, `NEWS` & `README.md`. I'll leave that change to you for when/if you release these changes.

If it helps, I noted all of the changed files in the first comment.

Kogiku committed:
Compile with sassc

Kogiku committed:
fix gnome-shell 3.30+ -slider-handle

Kogiku commented:
I put 'WIP' in the title because I realized accidentally pushed a non gulp regenerated version. As a result, none of these changes would have taken affect. I decided to fix this by making this theme compile with sassc (another change from Arc).

Kogiku committed:
remove all gulp/node-sass files

* [additional edit] HACKING.md: better documentation

Kogiku commented:
Oops, you my want to hold off or cancel this pull. By removing the css files, I've removed all changes you and echuber2 made to gtk-dark.css, gtk-light.css, and gtk.css files. All of those changes should have gone in the sass/*.scss files. I will migrate all of those changes, but it might take a while.

Kogiku committed:
AUTHORS: Arc Theme contributors url for consistency

Kogiku committed:
gnome-shell 3.30+: clean code for overview page indicators

Kogiku committed:
Makefile.am: $(srcdir) unnecessary

Kogiku committed:
gnome-shell: revert font back to 9pt

I don't feel comfortable making this disign decision for this theme. It looks fine to me either way, but it may look off to some GNOME users who were used to the way it was.

Kogiku commented:
Okay, I'm done poking at this theme.

Build changes:
* `--with-gnome-shell` replaces`--with-gnome`
* `sassc` is now a build depend for GTK3 and GNOME Shell

Kogiku committed:
AUTHORS: add jnsh's Arc Theme contributors URL

Kogiku committed:
build: Move Plank theme from extra to common

from https://github.com/jnsh/arc-theme/commit/b81a91fe9d238bd9e79299e0d503f36a4e281b24

Kogiku committed:
Set GTK3_VERSION to compatible version instead of actual version

from https://github.com/jnsh/arc-theme/commit/7a0d3be51708360746e74c1bae3fd9f7ba5c95d0

Kogiku committed:
gnome-shell: separate 3.22(3.20 base) 3.24(3.20), 3.28(3.26), 3.34(3.32), symlink 3.20 to 3.18

* assets symlinked to ../3.22/assets

Kogiku committed:
gnome-shell: Remove message-list-section clear button

from https://github.com/jnsh/arc-theme/commit/a7a400fca154cf9415d6430bf3c28615cb0c267c

Kogiku committed:
gnome-shell: Remove message-list-section-title definitions

from https://github.com/jnsh/arc-theme/commit/ac71a8f7daf238f025813dd43431ca943286218a

addendum to https://github.com/Kogiku/vertex-theme/commit/f8c6a4d322d83a5b41c20ea38a6ff9f2799f444e

Kogiku committed:
gnome-shell: Add Pad OSD theming

from https://github.com/jnsh/arc-theme/commit/12955c94ff773183f68c98bf7f7195839fa34836

Kogiku committed:
gnome-shell: Add spacing between dateMenu displays

from https://github.com/jnsh/arc-theme/commit/b2fc62233935265b9473f27fca9c43571c4046d6

Kogiku committed:
gnome-shell: make notification icons prefer symbolic variants

from https://gitlab.gnome.org/GNOME/gnome-shell/commit/d3c050b88d497e87eb54bb3787a122cd282a89f6

Kogiku committed:
gnome-shell: Fix nesting of elements

from https://github.com/jnsh/arc-theme/commit/756b1adc6cb9149f124d6098a6ef567c08d6c2cd

Kogiku committed:
gnome-shell: theme weather button

Kogiku committed:
gnome-shell: Use font-relative width for prompts

from https://github.com/jnsh/arc-theme/commit/48b7b86fb69dc23e6dfa382733da8f29ed937a42

Kogiku committed:
gnome-shell: loginDialog: Only highlight a single item at any time

from https://github.com/jnsh/arc-theme/commit/ee125996e9ddb4acde9f596cd695925a9b642049

Kogiku committed:
gnome-shell: theme weather button (addendum to https://github.com/Kogiku/vertex-theme/commit/785c3c959a2b03267e6b8a769400f5a6ef3b7832)

Kogiku committed:
gnome-shell: fix calendar and font

undo changes from https://github.com/arc-design/arc-theme/commit/6bd20fff10f43cbc7cc5587a5dad7a100023d180 & https://github.com/arc-design/arc-theme/commit/6845a0544ff233033745f442110a5e47a2dbaf1d

Kogiku committed:
gnome-shell: More compact search results

from https://github.com/jnsh/arc-theme/commit/fb6db51e4cb2eb9eede30390377d68670eafe684

Kogiku committed:
gnome-shell: Refactor search provider info

from https://github.com/jnsh/arc-theme/commit/dcb9fbacde3ca7239cd6c1b9cada6065691979ca

Kogiku committed:
gnome-shell: Use standard font size for search results titles

from https://github.com/jnsh/arc-theme/commit/683b5665e41a58bf3f77103fccd9b58fc4b4d605

Kogiku committed:
gnome-shell: Add transparency to search result descriptions

from https://github.com/jnsh/arc-theme/commit/bd221715c6afbd5590003eb13264b7ef7a22621c

Kogiku committed:
gnome-shell: Update mount dialog classes

from https://github.com/jnsh/arc-theme/commit/1e34cf78fb7fb5875c4d26b619ce7efb226d8007

Kogiku committed:
gnome-shell: Add message dialog style classes

from https://github.com/jnsh/arc-theme/commit/0bced4b36d6a4dfff3ff88702f44d1c952cb0b91

Kogiku committed:
gnome-shell: Add extension-dialog style classes

from https://github.com/jnsh/arc-theme/commit/6dbda5cb57bd2199e55c67ee659cc38866de3b7c

Kogiku committed:
gnome-shell: Add max-width to .mount-dialog-subject

accidentally removed in commit https://github.com/Kogiku/vertex-theme/commit/427f9fb33e0c73927fbdeb932418fe317e86c088

Kogiku committed:
gnome-shell: Use .message-dialog-* classes

from https://github.com/jnsh/arc-theme/commit/b7a939447e76837b734faac79690f5c53cb78590

Kogiku committed:
gnome-shell: Add inhibit-shortcuts-dialog class

from https://github.com/jnsh/arc-theme/commit/d576d3506914cd67d54cda9224a44c86e917b0d8

Kogiku committed:
gnome-shell: fix activities overview close button

Kogiku committed:
remove ChangeLog/NEWS files

Changelogs are nice, but I found it to be a pain to keep up with for this project.

Kogiku committed:
gnome-shell: remove redundant :hover selector from .window-caption

from https://gitlab.gnome.org/GNOME/gnome-shell-sass/commit/3ce23ca9648c131a33b79f37b3e19001ff45450d

Kogiku committed:
gnome-shell: Reduce window preview spacing

from https://github.com/jnsh/arc-theme/commit/5ee499d098865aad3f173b0e1351f60e63b68d91

Kogiku committed:
gnome-shell: Remove .legacy-tray class

from https://github.com/jnsh/arc-theme/commit/12528ca188808b372f985eb8d1df757fe2067cf8

Kogiku committed:
gnome-shell: Add system action icon theming

from https://github.com/jnsh/arc-theme/commit/99dac49526c153ecd3ba51b07166cd761546f361

Kogiku committed:
gnome-shell: Add workspace-thumbnail placeholder

from https://github.com/jnsh/arc-theme/commit/bc19cf29255a3d26f72480fe0b2f8cf93ebc9c7d

Kogiku committed:
gnome-shell: Align system-sub-menus with parent

from https://github.com/jnsh/arc-theme/commit/d5d651a517be0ba14a3e83ed2995f44e343a6541

Kogiku committed:
gnome-shell: osd barlevel fix/tweaks

the backgroung image does not fit the new barlevel here and I accidently left in the old padding.

Kogiku committed:
gnome-shell: fix workspace-thumbnail placeholder asset

Kogiku committed:
gnome-shell: Update OSK theme

based on https://github.com/jnsh/arc-theme/commit/1d1fa4aee95e483128001a3ea57c34d76f7bdf02 & https://github.com/jnsh/arc-theme/commit/c7a8e929769ad3b47bd79fcc4a66d13bc508f652

Kogiku committed:
Merge pull request #1 from Kogiku/port-arc-jnsh-changes

Port various changes from the Arc Theme (jnsh's fork)

Kogiku committed:
gnome-shell: Update login user-list

from https://github.com/jnsh/arc-theme/commit/d119b821f49184bba7af46a81135a01a681bec28

Kogiku committed:
gnome-shell: Use tabular figures for some numbers

from https://github.com/jnsh/arc-theme/commit/03f1698c4be65640cc53ff7e8a8d27e882df00c1

Kogiku committed:
gnome-shell: Add remote-access-indicator theming

from https://github.com/jnsh/arc-theme/commit/b738988ec4feb1925f9c672cdd47f0a86fca181d

Kogiku committed:
gnome-shell: Use symbolic icons in action buttons

from https://github.com/jnsh/arc-theme/commit/c545ea57dc8d5a95e4ee5004641e349ad8f57150

Kogiku committed:
gnome-shell: Replace page indicator & corner ripple SVGs with CSS

from https://github.com/jnsh/arc-theme/commit/88cd4b955031436c505a112b41e2b71a44242ea5 & https://github.com/jnsh/arc-theme/commit/34f896c330ff6ecd841f06948b96de26f0f28f36

Kogiku committed:
gnome-shell: gnome-shell: Update media fallback cover size

from https://github.com/jnsh/arc-theme/commit/5e6c5f8731df97648bef89c1549b0a01a3fbd281

Kogiku committed:
gnome-shell: Update notification style

based on https://github.com/jnsh/arc-theme/commit/576160b7a720eaa168c6807ad04c3ddb3a3b0269 & https://github.com/jnsh/arc-theme/commit/54362518295adb8429e1d0e3b51636c6eab243c3 & https://github.com/jnsh/arc-theme/commit/5e6e7bc220acc3d4b7836e6dba8a2d42a808401b

Kogiku committed:
gnome-shell: Update dateMenu

from https://github.com/jnsh/arc-theme/commit/fed65a8b91087a5592c93f56caced0fc75601c02

Kogiku committed:
gnome-shell: Restrict app menu width

from https://github.com/jnsh/arc-theme/commit/d49a30e4fa37eb2df0f9049283d30f00fed04927

Kogiku committed:
gnome-shell: update switch buttons

Kogiku committed:
gnome-shell: Handle fallback notification icons and refine code

from https://github.com/jnsh/arc-theme/commit/70243f6a4cfe9114886bfbba1f77d3d5ae0cb3e1 & addendum to https://github.com/Kogiku/vertex-theme/commit/86ce0ff3dd32515470ac2fce66f5d26a6382670c

Kogiku committed:
gnome-shell: Update popup-menu-arrow

from https://github.com/jnsh/arc-theme/commit/e1ad56b167b38ad618d3bcfb5b350a6205cef713

Kogiku committed:
gnome-shell: Update prompt-dialog-error-label

from https://github.com/jnsh/arc-theme/commit/2de712ed22596d48972741bb830bd56666443cce

Kogiku committed:
gnome-shell: Add select-area-rubberband theming

from https://github.com/jnsh/arc-theme/commit/4b73e328b2d9c3288f4c4514516fd00d41c779ac

Kogiku committed:
gnome-shell: Use em for .calendar-day-base sizing

from https://github.com/jnsh/arc-theme/commit/5922658f7b7358aa597d1dab5ce7027ea5dff850

Kogiku committed:
gnome-shell: Fix a number of typos

from https://github.com/jnsh/arc-theme/commit/e292959472cc683ed45363274f58863c2a17b65e

Kogiku committed:
gnome-shell: Calendar arrows as buttons.

Kogiku committed:
gnome-shell: Refine calendar arrow buttons

addendum to https://github.com/Kogiku/vertex-theme/commit/8ea8d61c1a9bba3bcc0dd2f2f054461794b3e7ff

Kogiku committed:
gnome-shell: Remove message-active.svg

addendum to https://github.com/Kogiku/vertex-theme/commit/86ce0ff3dd32515470ac2fce66f5d26a6382670c

Kogiku committed:
gnome-shell: Remove redundant border property in scrollbar handle

from https://github.com/jnsh/arc-theme/commit/5b805de194d7226eab8af1c57c08784b779ba613

Kogiku committed:
gnome-shell: Add new prompt-dialog classes

from https://github.com/jnsh/arc-theme/commit/99185f3ef81d71194ce38032744d2518c6a941f1

Kogiku committed:
gnome-shell: Add message close button styling

from https://github.com/jnsh/arc-theme/commit/b14dc452519341240580e383c4f7ea7450f13b59

Kogiku committed:
gnome-shell: Drop appMenu spinner-image

from https://github.com/jnsh/arc-theme/commit/c99efe3ce51d39d143eefbafe97bfd5dbb9ec6ec

Kogiku committed:
gnome-shell: Add horizontal ws-switcher styling

from https://github.com/jnsh/arc-theme/commit/f69565f18a58d4f58c6f37248c8294bc77215351

Kogiku committed:
gnome-shell: Update search results stylings

from https://github.com/jnsh/arc-theme/commit/dca94df86f92476864918ae02a032c45406b680e

Kogiku committed:
gnome-shell: Undo calendar arrows as buttons

I decided to restore them based on an old 3.14 screenshot. [addendum to https://github.com/Kogiku/vertex-theme/commit/8ea8d61c1a9bba3bcc0dd2f2f054461794b3e7ff & https://github.com/Kogiku/vertex-theme/commit/3091080be1d9fdb19d22c70c4f3b926550784f04]

Kogiku committed:
gnome-shell: Add folder rename popup styling

from https://github.com/jnsh/arc-theme/commit/492f74d5c3b442d265c0942da05b24cfda451028

Kogiku committed:
gnome-shell: Add drag-over style for app folders

from https://github.com/jnsh/arc-theme/commit/75c3f240a96cabd4d022e95a7a3b6da0da04820a

Kogiku committed:
build: improve *.m4 files for versioned themes

from https://github.com/jnsh/arc-theme/commit/7b36b73cb3e09cfd39b33d698b6998b274babb87 & https://github.com/jnsh/arc-theme/commit/71143945f952b9645d3151bd90aadf6be4df8d9d & https://github.com/jnsh/arc-theme/commit/f0912fc0fc091094241bbdbdba7437bf8c93723c & https://github.com/jnsh/arc-theme/commit/45783e220e3b47bd52c2fef091043fd4caae8f5b

Kogiku committed:
Merge pull request #2 from Kogiku/port-arc-jnsh-changes

Port more changes from the Arc Theme (jnsh's fork)

Kogiku committed:
gnome-shell: Add pointer-location ripple styling

from https://github.com/jnsh/arc-theme/commit/f0d6275e1991c8c45b7b2f40c99312ab765bb6ad
